code/modules/events/unknown.dm

```diff
@@ -0,0 +1,47 @@
+/datum/event/spawn_unknown
+	var/key_of_unknown
+
+/datum/event/spawn_unknown/proc/get_unknown(var/end_if_fail = 0)
+	spawn()
+		var/list/candidates = pollCandidates("Do you want to play as an unknown entity?", ROLE_DEMON, 1, 300, 0, 24)
+		if(!candidates.len)
+			key_of_unknown = null
+			return kill()
+		var/mob/C = pick(candidates)
+		key_of_unknown = C.key
+
+		if(!key_of_unknown)
+			return kill()
+
+		var/datum/mind/player_mind = new /datum/mind(key_of_unknown)
+		player_mind.active = 1
+		var/list/spawn_locs = list()
+		for(var/obj/effect/landmark/L in GLOB.landmarks_list)
+			if(isturf(L.loc))
+				switch(L.name)
+					if("revenantspawn")
+						spawn_locs += L.loc
+		if(!spawn_locs) //If we can't find any revenant spawns, try the carp spawns
+			for(var/obj/effect/landmark/L in GLOB.landmarks_list)
+				if(isturf(L.loc))
+					switch(L.name)
+						if("carpspawn")
+							spawn_locs += L.loc
+		if(!spawn_locs) //If we can't find either, just spawn the revenant at the player's location
+			spawn_locs += get_turf(player_mind.current)
+		if(!spawn_locs) //If we can't find THAT, then just retry
+			return kill()
+		var/obj/effect/dummy/thing/holder = new /obj/effect/dummy/thing(pick(spawn_locs))
+		var/mob/living/simple_animal/hostile/thing/T = new /mob/living/simple_animal/hostile/thing(holder)
+		T.holder = holder
+		T.phasing = TRUE
+		player_mind.transfer_to(T)
+		player_mind.assigned_role = "Slaughter Demon"
+		player_mind.special_role = SPECIAL_ROLE_SLAUGHTER_DEMON // "basically" the same thing
+		message_admins("[key_name_admin(T)] has been made into The Unknown by an event.")
+		log_game("[key_name_admin(T)] was spawned as The Unknown by an event.")
+		world << sound('sound/thing/warning.ogg')
+		return 1
+
+/datum/event/spawn_unknown/start()
+	get_unknown()
```

Comment:
Try looking at abductor code in order to get an idea of how candidate polling should work